src/EventStore.Core/TransactionLog/ITransactionFileWriter.cs

```diff
@@ -6,9 +6,12 @@ namespace EventStore.Core.TransactionLog {
 	public interface ITransactionFileWriter : IDisposable {
 		void Open();
 		bool Write(ILogRecord record, out long newPos);
+		void Commit();
```

Comment:
We could consider having something like a Begin that just checks that there aren't already uncommitted writes (that might therefore end up being committed accidentally as part of a separate transaction)

Reply:
great idea :+1: i've pushed the required changes